packages/calcite-components/src/components/sort-handle/sort-handle.tsx

```diff
@@ -0,0 +1,452 @@
+    });
+  };
+
+  private handleMoveTo = (event: Event): void => {
+    const id = (event.target as HTMLElement).dataset.id;
+    const moveTo = this.moveToItems.find((item) => item.id === id);
+    this.calciteSortHandleMove.emit({ moveTo });
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const {
+      disabled,
+      flipPlacements,
+      messages,
+      open,
+      overlayPositioning,
+      placement,
+      scale,
+      setPosition,
+      setSize,
+      widthScale,
+    } = this;
+    const text = this.getLabel();
+
+    const isDisabled = disabled || !setPosition || !setSize;
+
+    return (
+      <InteractiveContainer disabled={disabled}>
+        <calcite-dropdown
+          class={CSS.dropdown}
+          disabled={isDisabled}
+          flipPlacements={flipPlacements}
+          onCalciteDropdownBeforeClose={this.handleBeforeClose}
+          onCalciteDropdownBeforeOpen={this.handleBeforeOpen}
+          onCalciteDropdownClose={this.handleClose}
+          onCalciteDropdownOpen={this.handleOpen}
+          overlayPositioning={overlayPositioning}
+          placement={placement}
+          ref={this.setDropdownEl}
+          scale={scale}
+          widthScale={widthScale}
+        >
+          <calcite-action
+            active={open}
+            appearance="transparent"
+            class={CSS.handle}
+            icon={disabled ? ICONS.blank : ICONS.drag}
+            label={text}
+            scale="s"
+            slot="trigger"
+            text={text}
+            title={text}
+          />
+          <calcite-dropdown-group
+            groupTitle={messages.reorder}
+            key="reorder"
+            scale={scale}
+            selectionMode="none"
+          >
+            {this.renderTop()}
+            {this.renderUp()}
+            {this.renderDown()}
+            {this.renderBottom()}
+          </calcite-dropdown-group>
+          {this.renderMoveToGroup()}
+        </calcite-dropdown>
+      </InteractiveContainer>
+    );
+  }
+
+  private renderMoveToItem(moveToItem: MoveTo): VNode {
+    return (
+      <calcite-dropdown-item
+        data-id={moveToItem.id}
+        key={moveToItem.id}
+        label={moveToItem.label}
+        onCalciteDropdownItemSelect={this.handleMoveTo}
+      >
+        {moveToItem.label}
+      </calcite-dropdown-item>
+    );
+  }
+
+  private renderMoveToGroup(): VNode {
+    const { messages, moveToItems, scale } = this;
+
+    return moveToItems?.length ? (
+      <calcite-dropdown-group
+        groupTitle={messages.moveTo}
+        key="move-to-items"
+        scale={scale}
+        selectionMode="none"
+      >
+        {moveToItems.map((moveToItem) => this.renderMoveToItem(moveToItem))}
+      </calcite-dropdown-group>
+    ) : null;
+  }
+
+  private renderTop(): VNode {
```

Comment:
There's some duplicate logic here we can simplify. Could you try something like this?

```tsx
private renderDropdownItem(positionIndex: number, label: string): VNode {
  const { messages } = this;

  return (
    <calcite-dropdown-item
      data-value={REORDER_VALUES[positionIndex]}
      key={REORDER_VALUES[positionIndex]}
      label={label}
      onCalciteDropdownItemSelect={this.handleReorder}
    >
      {label}
    </calcite-dropdown-item>
  );
}

private renderTop(): VNode | null {
  const { setPosition, messages } = this;
  return setPosition !== 1 && setPosition !== 2 ? this.renderDropdownItem(0, messages.moveToTop) : null;
}

private renderUp(): VNode | null {
  const { setPosition, messages } = this;
  return setPosition !== 1 ? this.renderDropdownItem(1, messages.moveUp) : null;
}

private renderDown(): VNode | null {
  const { setPosition, setSize, messages } = this;
  return setPosition !== setSize ? this.renderDropdownItem(2, messages.moveDown) : null;
}

private renderBottom(): VNode | null {
  const { setPosition, setSize, messages } = this;
  return setPosition !== setSize && setPosition !== setSize - 1 ? this.renderDropdownItem(3, messages.moveToBottom) : null;
}```